crates/op-rbuilder/src/revert_protection.rs

```diff
@@ -19,19 +25,67 @@ pub trait EthApiOverride {
     async fn send_bundle(&self, tx: Bundle) -> RpcResult<BundleResult>;
 }
 
-pub struct RevertProtectionExt<Pool, Provider> {
+#[rpc(server, client, namespace = "eth")]
+pub trait EthApiOverrideReplacement<R: RpcObject> {
+    // Name TBD
+    #[method(name = "getTransactionReceipt")]
+    async fn transaction_receipt(&self, hash: B256) -> RpcResult<Option<R>>;
+}
+
+pub struct RevertProtectionExt<Pool, Provider, Eth, Network = op_alloy_network::Optimism> {
     pool: Pool,
     provider: Provider,
+    eth_api: Eth,
+    _network: std::marker::PhantomData<Network>,
 }
 
-impl<Pool, Provider> RevertProtectionExt<Pool, Provider> {
-    pub fn new(pool: Pool, provider: Provider) -> Self {
-        Self { pool, provider }
+impl<Pool, Provider, Eth, Network> RevertProtectionExt<Pool, Provider, Eth, Network>
+where
+    Pool: Clone,
+    Provider: Clone,
+    Eth: Clone,
+{
+    pub fn new(pool: Pool, provider: Provider, eth_api: Eth) -> Self {
+        Self {
+            pool,
+            provider,
+            eth_api,
+            _network: std::marker::PhantomData,
+        }
+    }
+
+    pub fn bundle_api(&self) -> RevertProtectionBundleAPI<Pool, Provider> {
+        RevertProtectionBundleAPI {
+            pool: self.pool.clone(),
+            provider: self.provider.clone(),
+        }
     }
+
+    pub fn eth_api(&self, reverted_cache: SharedLruCache<B256, ()>) -> RevertProtectionEthAPI<Eth> {
+        RevertProtectionEthAPI {
+            eth_api: self.eth_api.clone(),
+            reverted_cache,
+        }
+    }
+}
+
+pub type SharedLruCache<K, V> = Arc<Mutex<LruCache<K, V>>>;
+
+pub fn create_shared_cache<K, V>(capacity: usize) -> SharedLruCache<K, V>
+where
+    K: std::hash::Hash + Eq,
+{
+    let cache = LruCache::new(NonZeroUsize::new(capacity).unwrap());
+    Arc::new(Mutex::new(cache))
+}
```

Comment:
I'd suggest using [Moka](https://crates.io/crates/moka) instead of this because if we use Arc-Mutex we will be only serving one RPC call at a time and won't have scalable/concurrent RPC API.